app/src/pages/SingleContent.vue

```diff
@@ -22,9 +23,28 @@ const content = db.getBySlugAsRef<ContentDto>(props.slug);
 const tagsContent = ref<ContentDto[]>([]);
 
 watch(
-    content,
+    [content, appLanguageAsRef],
     async () => {
-        if (!content.value) return;
+        if (!content.value) {
+            return;
+        }
+
+        if (appLanguageAsRef.value?._id != content.value.language) {
+            const contentDocs = await db.whereParent(content.value.parentId);
+            const preferred = contentDocs.find((c) => c.language == appLanguageAsRef.value?._id);
+
+            if (preferred) {
+                await router.push({ name: "content", params: { slug: preferred.slug } });
```

Comment:
Instead of `router.push`, `router.replace` might be a good option as well. It will replace the current URL, so when you click back in the browser, you don't go back to the "wrong" language, but instead to the page before that, which might be the homepage.